### ISSUE TITLE
Since Unity 5.3.4p2 UI canvas is not displayed

### ISSUE THREAD

**Issue by KuRouZu** [googlevr/gvr-unity-sdk#211]
Tested with sample scene and cardboard version 0.7.0 and 0.7.2. Until the Unity version 5.3.4p1 is working fine. I've also tested with version 5.3.4p4 and not working. 
Since the p2 version, unity reintroduced the particles batching, so I'm stuck in the latest versions of unity or else I get very low framerates. Is there any workaround to fix this?

Regards.

**Comment by smdol** [comment]
I did some testing with just using a simple two-camera setup, one pointed at a uGUI Canvas, and rendering to a target RenderTexture, which is then applied to a Plane object. The second camera is pointed at the Plane.  The canvas is not showing up in the RenderTexture at all, which seems to be the root cause of the issue.

We use this "render to RenderTexture, display it on a mesh" technique to do distortion correction, so that's why you see the issue.  There must be a regression in Unity, or perhaps there are changes to how we have to render to a texture in order for uGUI to work.  I don't see any hints about this in their release notes, so it might be a Unity bug.

**Comment by yalfers** [comment]
Also came across this issue. So far it has been an issue from 5.3.4p2+ for me.

**Comment by smdol** [comment]
You can also get the issue to not happen by setting Distortion to None.  In either case, Cardboard is not in the render path, so this problem does not surface.

**Comment by KuRouZu** [comment]
Still happening in unity 5.3.4p5. I have found this reported in unity issue tracker related to world space canvas and rendertexture. It may be related to this problem.
 [https://issuetracker.unity3d.com/issues/ui-elements-arent-rendered-to-rendertexture](https://issuetracker.unity3d.com/issues/ui-elements-arent-rendered-to-rendertexture)

**Comment by nathanmartz** [comment]
Yes. It appears to be unrelated to Cardboard but that deeper UI elements on
render textures bug.

**Comment by nosmirck** [comment]
Confirmed to be happening still in 5.3.5f1.

For those that are still having trouble with this, I kind of found a workaround.

I duplicated the 2 cameras (eyes) and removed the GvrEye Script on both, enabled them, set the culling mask to WorldUI (a custom layer) and Depth to 100.

On the GvrPostRender Script, I changed the depth to 99 instead of 100. This allows me to render the World canvas on top everything.

As my world canvas was used to show some HUD elements, it worked perfectly. Of course, you don't get any distortion on the canvas, but, I used the Curved UI plugin and it did the trick for me.

Let's hope Unity priorize this issue and gets resolved soon.
